substrate/client/service/src/client/client.rs

```diff
@@ -846,11 +848,13 @@ where
 					Block::new(import_block.header.clone(), body.clone()),
 				)?;
 
-				let state = self.backend.state_at(*parent_hash)?;
+				let state = self.backend.state_at(*parent_hash, Some(CallContext::Onchain))?;
```

Comment:
Maybe it can be better to introduce a new method for this particular place?

Reply:
Added an enum.